addons/mail/static/src/discuss/typing/common/composer_patch.js

```diff
@@ -42,6 +42,14 @@ patch(Composer.prototype, {
             );
         }
     },
+    initateNotifyIsTyping(value) {
+        if (!this.typingNotified && value) {
+            this.typingNotified = true;
+            this.notifyIsTyping();
+            browser.setTimeout(() => (this.typingNotified = false), LONG_TYPING);
+        }
+        this.stopTypingDebounced();
+    },
```

Comment:
I don't like the name of function nor the split.

`onInput` method of `Composer` exists for the sole purpose of stating whether self is typing or not. `ev.target.value` is simply the textarea value, so we could just use `this.ref.el.value` in the end, killing de dependence on a parameter on this method. By doing so, the fix is simply renaming`onInput` to `detectTyping`, using `this.ref.el.value`, and call it after `addEmoji()`

Reply:
Hello, I tried this, but it didn't work because `addEmoji` doesn't wait for the emoji to actually be added to the composer. As a result, `this.notifyIsTyping` is not triggered even if we add an emoji to an empty composer, since `this.ref.el.value` will be an empty string by the time it reaches the `if (!this.typingNotified && value)` check.

Reply:
It changes the model, so this needs to wait next rendering since this is a `t-model`.

I wonder if we can use `composer.text` instead of `this.ref.el.value`, hopefully `onInput` sees the updated value in model...

Reply:
I tested it and all test pass including yours if we use `composer.text` instead of `ref.el.value`. I'm push on the branch with the change